autopush/websocket.py

```diff
@@ -366,14 +369,27 @@ def _check_other_nodes(self, result):
             node_id = self._get_aval(attrs.get("node_id", {}), 'S')
             uaid = self._get_aval(attrs.get("uaid", {}), 'S')
             connected_at = self._get_aval(attrs.get("connected_at", {}), 'N')
-            url = "%s/notif/%s/%s" % (node_id, uaid, connected_at)
-            if not getattr(self, 'testing', False):
-                d = self.ap_settings.agent.request(
-                    "DELETE",
-                    url.encode("utf8"),
-                ).addCallback(self.finish_hello)
-                d.addErrback(self.log_err, extra="Failed to delete old node")
-                return
+            inSelf = self.ap_settings.clients.get(uaid)
+            if inSelf:
+                if inSelf.connected_at <= connected_at:
+                    inSelf.sendClose()
+            else:
+                url = "%s/notif/%s/%s" % (node_id, uaid, connected_at)
+
+                def _eatConnections(*args):
+                    failure.trap(
+                        ConnectError, ConnectionRefusedError, UserError
+                    )
+
+                if not getattr(self, 'testing', False):
+                    self.ap_settings.agent.request(
+                        "DELETE",
+                        url.encode("utf8"),
+                    ).addErrback(
+                        _eatConnections
+                    ).addErrback(
+                        self.log_err,
+                        extra="Failed to delete old node")
```

Comment:
The rationale for having each callback/Errback be a separate line is similar to the rule on imports. That is, as separate lines they're easier to see what is changing in a diff. It's why once you import more than a few things from a module, its generally better to go multiline with a (), so that changes to imports are easier to parse in a diff than modifications to a single line.

I.e.
```
from something import foo, smith, george, fred

# instead should be

from something import (
    foo,
    smith,
    george,
    fred,
)
```